src/gc/unix/cgroup.cpp

```diff
@@ -0,0 +1,248 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*++
+
+Module Name:
+
+    cgroup.cpp
+
+Abstract:
+    Read memory limits for the current process
+--*/
+
+#include "cgroup.h"
+
+CGroup::CGroup()
+{
+    if(!findMemoryHierarchyMount(_memory_cgroup_path, MAX_PATH))
+    {
+        _memory_cgroup_path[0] = '\0';
+        return;
+    }
+    char cgroup_path_relative_to_mount[MAX_PATH];
+    if(!findCGroupPathForMemorySubsystem(cgroup_path_relative_to_mount, MAX_PATH))
+    {
+        _memory_cgroup_path[0] = '\0';
+        return;
+    }
+    strcat_s(_memory_cgroup_path, MAX_PATH, cgroup_path_relative_to_mount);
+}
+
+bool CGroup::physicalMemoryLimit(size_t &val)
+{
+    char mem_limit_filename[MAX_PATH];
+
+    if(_memory_cgroup_path[0] == 0)
+        return false;
+
+    strcpy_s(mem_limit_filename, MAX_PATH, _memory_cgroup_path);
+    strcat_s(mem_limit_filename, MAX_PATH, MEM_LIMIT_FILENAME);
+    return readMemoryValueFromFile(mem_limit_filename, val);
+}
+
+bool CGroup::swapMemoryLimit(size_t &val)
+{
+    char memsw_limit_filename[MAX_PATH];
+
+    if(_memory_cgroup_path[0] == 0)
+        return false;
+
+    strcpy_s(memsw_limit_filename, MAX_PATH, _memory_cgroup_path);
+    strcat_s(memsw_limit_filename, MAX_PATH, MEM_SWAP_LIMIT_FILENAME);
+    return readMemoryValueFromFile(memsw_limit_filename, val);
+}
+
+bool CGroup::findMemoryHierarchyMount(char* mountpath, size_t mountpathsize)
+{
+    bool retval = false;
+    char *line = nullptr;
+    size_t lineLen = 0;
+    FILE *mountinfofile = fopen(PROC_MOUNTINFO_FILENAME, "r");
+    if(mountinfofile == NULL)
+        goto done;
+
+    while(!retval && getline(&line, &lineLen, mountinfofile) != -1)
+    {
+        char filesystemType[100];
+        char options[MAX_PATH];
+
+        char* separatorChar = strchr(line, '-');
+
+        // See man page of proc to get format for /proc/self/mountinfo file
+        int sscanfRet = sscanf_s(separatorChar, 
```

Comment:
the sscanf_s and other _s functions are not available for plain Unix. You'll need to use the plain ones or maybe better add defines that define sscanf_s as sscanf etc so that the rest of the code is the same as for the PAL version.